knotx-server/src/main/java/io/knotx/server/SupportedMethodsAndPathsHandler.java

```diff
@@ -38,24 +44,31 @@ public static SupportedMethodsAndPathsHandler create(KnotxServerConfiguration co
 
   @Override
   public void handle(RoutingContext context) {
-    boolean shouldRejectMethod = configuration.getEngineRouting().keySet().stream()
-        .noneMatch(supportedMethod -> supportedMethod == context.request().method());
-
-    boolean shouldRejectPath = configuration.getEngineRouting().values().stream().noneMatch(
-        routingEntries -> routingEntries.stream()
-            .anyMatch(item -> context.request().path().matches(item.path()))
-    );
-
-    if (shouldRejectMethod) {
-      LOGGER.warn("Requested method {} is not supported based on configuration",
-          context.request().method());
-      context.fail(HttpResponseStatus.METHOD_NOT_ALLOWED.code());
-    } else if (shouldRejectPath) {
-      LOGGER.warn("Requested path {} is not supported based on configuration",
-          context.request().path());
+    String path = context.request().path();
+    HttpMethod method = context.request().method();
+    boolean pathNotAllowedInDefaultFlow = isPathNotPresentInFlow(path, defaultFlow);
+    boolean pathNotAllowedInCustomFlow = isPathNotPresentInFlow(path, customFlow);
+    if (pathNotAllowedInDefaultFlow && pathNotAllowedInCustomFlow) {
+      LOGGER.warn("Requested path {} is not supported based on configuration", path);
       context.fail(HttpResponseStatus.NOT_FOUND.code());
+    } else if (!pathNotAllowedInDefaultFlow && isMethodDisallowedInFlow(method, defaultFlow)) {
```

Comment:
So.. what exactly should happen, when `defaultPath` and `defaultMethod` are not allowed, but `customPath` and `customMethod` are allowed :) ?
It looks that we finish without any action on context (`tip`: there is double `if`).